src/main/java/nextstep/subway/path/domain/PathFinder.java

```diff
@@ -0,0 +1,88 @@
+package nextstep.subway.path.domain;
+
+import lombok.Builder;
+import lombok.Getter;
+import nextstep.subway.common.exception.BadRequestException;
+import nextstep.subway.line.domain.Line;
+import nextstep.subway.line.domain.Sections;
+import nextstep.subway.station.domain.Station;
+import org.jgrapht.GraphPath;
+import org.jgrapht.alg.shortestpath.DijkstraShortestPath;
+import org.jgrapht.graph.DefaultWeightedEdge;
+import org.jgrapht.graph.WeightedMultigraph;
+
+import java.util.List;
+import java.util.Objects;
+
+public class PathFinder {
+
+    public static final String SAME_STATION_MESSAGE = "출발역과 도착역이 같습니다.";
+    public static final String NON_EXIST_PATH_MESSAGE = "출발역과 도착역이 연결되어 있지 않습니다.";
+
+    private PathFinderRequest request;
+
+    private PathFinder(PathFinderRequest request) {
+        this.request = request;
+    }
+
+    public static PathFinder of(PathFinderRequest request) {
+        return new PathFinder(request);
+    }
+
+    public GraphPath searchShortestPath() {
+        validateFindBefore(request);
+        WeightedMultigraph<Station, DefaultWeightedEdge> graph = getGraphByRequest(request);
+
+        DijkstraShortestPath dijkstraShortestPath = new DijkstraShortestPath(graph);
+        GraphPath graphPath = dijkstraShortestPath.getPath(request.startStation, request.endStation);
+        validateFindAfter(graphPath);
+
+        return graphPath;
+    }
```

Comment:
제네릭은 항상 명시화해주시면 좋아요! 👍 

```suggestion
    public GraphPath<Station, DefaultWeightedEdge> searchShortestPath() {
        validateFindBefore(request);
        WeightedMultigraph<Station, DefaultWeightedEdge> graph = getGraphByRequest(request);

        DijkstraShortestPath<Station, DefaultWeightedEdge> dijkstraShortestPath = new DijkstraShortestPath<>(graph);
        GraphPath<Station, DefaultWeightedEdge> graphPath = dijkstraShortestPath.getPath(request.startStation, request.endStation);

        validateFindAfter(graphPath);

        return graphPath;
    }

```